ngraph/core/src/op/non_max_suppression.cpp

```diff
@@ -530,3 +533,593 @@ void op::v4::NonMaxSuppression::validate_and_infer_types()
+    : Op({boxes,
+          scores,
+          max_output_boxes_per_class,
+          iou_threshold,
+          score_threshold,
+          soft_nms_sigma})
+    , m_box_encoding{box_encoding}
+    , m_sort_result_descending{sort_result_descending}
+    , m_output_type{output_type}
+{
+    constructor_validate_and_infer_types();
+}
+
+shared_ptr<Node>
+    op::v5::NonMaxSuppression::clone_with_new_inputs(const OutputVector& new_args) const
+{
+    check_new_args_count(this, new_args);
+    NODE_VALIDATION_CHECK(this,
+                          new_args.size() >= 2 && new_args.size() <= 6,
+                          "Number of inputs must be 2, 3, 4, 5 or 6");
+
+    const auto& arg2 = new_args.size() > 2
+                           ? new_args.at(2)
+                           : ngraph::op::Constant::create(element::i64, Shape{}, {0});
+    const auto& arg3 = new_args.size() > 3
+                           ? new_args.at(3)
+                           : ngraph::op::Constant::create(element::f32, Shape{}, {.0f});
+    const auto& arg4 = new_args.size() > 4
+                           ? new_args.at(4)
+                           : ngraph::op::Constant::create(element::f32, Shape{}, {.0f});
+    const auto& arg5 = new_args.size() > 5
+                           ? new_args.at(5)
+                           : ngraph::op::Constant::create(element::f32, Shape{}, {.0f});
+
+    return std::make_shared<op::v5::NonMaxSuppression>(new_args.at(0),
+                                                       new_args.at(1),
+                                                       arg2,
+                                                       arg3,
+                                                       arg4,
+                                                       arg5,
+                                                       m_box_encoding,
+                                                       m_sort_result_descending,
+                                                       m_output_type);
+}
+
+void op::v5::NonMaxSuppression::validate()
+{
+    const auto boxes_ps = get_input_partial_shape(0);
+    const auto scores_ps = get_input_partial_shape(1);
+
+    NODE_VALIDATION_CHECK(this,
+                          m_output_type == element::i64 || m_output_type == element::i32,
+                          "Output type must be i32 or i64");
+
+    if (boxes_ps.is_dynamic() || scores_ps.is_dynamic())
+    {
+        return;
+    }
+
+    NODE_VALIDATION_CHECK(this,
+                          boxes_ps.rank().is_static() && boxes_ps.rank().get_length() == 3,
+                          "Expected a 3D tensor for the 'boxes' input. Got: ",
+                          boxes_ps);
+
+    NODE_VALIDATION_CHECK(this,
+                          scores_ps.rank().is_static() && scores_ps.rank().get_length() == 3,
+                          "Expected a 3D tensor for the 'scores' input. Got: ",
+                          scores_ps);
+
+    if (inputs().size() >= 3)
```

Comment:
Looks like this check is always true, because you create fake inputs

Reply:
Done.